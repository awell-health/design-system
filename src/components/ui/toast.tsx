import * as React from "react"
import { cva, type VariantProps } from "class-variance-authority"

import { cn } from "@/lib/utils"
import { Icon } from "./icon";

const toastVariants = cva(
  "toast",
  {
    variants: {
      horizontal: {
        start: "toast-start",
        center: "toast-center",
        end: "toast-end",
      },
      vertical: {
        top: "toast-top",
        middle: "toast-middle",
        bottom: "toast-bottom",
      }
    },
    defaultVariants: {
      horizontal: "center",
      vertical: "bottom",
    },
  }
)

export interface ToastProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof toastVariants> {
      title: string;
      text?: string;
      icon?: JSX.Element;
      handleClose: () => void;
    }

function Toast({ className, horizontal, vertical, handleClose, ...props }: ToastProps) {
  return (
    <div className={cn(toastVariants({ horizontal, vertical }), className)}>
      <div className="p-4 bg-white rounded-lg shadow border border-slate-200 justify-start items-start gap-4 inline-flex max-w-[400px]">

        {props.icon && 
          <div className="relative"> 
            {props.icon}
          </div>
        }
        <div className="w-[400px] flex-col justify-start items-start">
          <div className="text-slate-900 text-sm font-medium leading-tight whitespace-normal overflow-auto max-w-[326px]">
            {props.title}
          </div>
          <div className="text-slate-500 text-sm leading-tight whitespace-normal overflow-aut">
            {props.text}
          </div>
        </div>

        <div role="button" onClick={handleClose}>
          <Icon icon="RiCloseFill" className="fill-slate-400"/>
        </div>
      </div>
    </div>
  )
}

export { Toast, toastVariants }
