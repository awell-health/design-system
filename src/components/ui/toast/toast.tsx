import * as React from 'react';
import { cva, type VariantProps } from 'class-variance-authority';

import { cn } from '@/lib/utils';
import { Icon } from '../icon';

const toastVariants = cva('toast', {
  variants: {
    horizontal: {
      start: 'toast-start',
      center: 'toast-center',
      end: 'toast-end'
    },
    vertical: {
      top: 'toast-top',
      middle: 'toast-middle',
      bottom: 'toast-bottom'
    }
  },
  defaultVariants: {
    horizontal: 'center',
    vertical: 'bottom'
  }
});

export interface ToastProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof toastVariants> {
  title: string;
  text?: string;
  icon?: JSX.Element;
  handleClose: () => void;
}

function Toast({ className, horizontal, vertical, handleClose, ...props }: ToastProps) {
  const { title, text, icon } = props;
  const textStyling = icon ? 'pl-4 flex items-start justify-center' : '';
  return (
    <div className={cn(toastVariants({ horizontal, vertical }), className)}>
      <div className='flex flex-col p-4 bg-white rounded-lg shadow border border-slate-200 max-w-[400px]'>
        <div className='justify-start items-center gap-4 inline-flex'>
          {icon && <div className='relative'>{icon}</div>}
          <div className='w-[400px] flex-col justify-start items-start'>
            <div className='text-slate-900 text-sm font-medium whitespace-normal overflow-auto max-w-[300px]'>
              {title}
            </div>
          </div>

          <div role='button' onClick={handleClose}>
            <Icon icon='RiCloseFill' className='fill-slate-400 w-4 h-4' />
          </div>
        </div>
        <div className={textStyling}>
          <div className='text-slate-500 text-sm whitespace-normal overflow-auto w-[300px]'>
            {text}
          </div>
        </div>
      </div>
    </div>
  );
}

export { Toast, toastVariants };
