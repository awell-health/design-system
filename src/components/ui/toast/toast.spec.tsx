import { fireEvent, render } from '@testing-library/react';
import { expect, it, describe, vi } from 'vitest';
import { Toast } from './toast';

describe('Toast', () => {
  const subject = (props = {}) => {
    const defaultProps = {
      title: 'Toast Title',
      handleClose: vi.fn(),
      text: 'Text',
      icon: <span>Icon</span>
    };

    return render(<Toast {...defaultProps} {...props} />);
  };

  it('match snapshot', () => {
    expect(subject()).toMatchSnapshot();
  });

  it('triggers handleClose function', () => {
    const handleClose = vi.fn();
    const { getByRole } = subject({ handleClose });

    fireEvent.click(getByRole('button'));

    expect(handleClose).toHaveBeenCalledOnce();
  });
});
