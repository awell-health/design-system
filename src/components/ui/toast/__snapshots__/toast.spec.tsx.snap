// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toast > match snapshot 1`] = `
<div>
  <div
    class="toast toast-center toast-bottom"
  >
    <div
      class="flex flex-col p-4 bg-white rounded-lg shadow border border-slate-200 max-w-[400px]"
    >
      <div
        class="justify-start items-center gap-4 inline-flex"
      >
        <div
          class="relative"
        >
          <span>
            Icon
          </span>
        </div>
        <div
          class="w-[400px] flex-col justify-start items-start"
        >
          <div
            class="text-slate-900 text-sm font-medium whitespace-normal overflow-auto max-w-[300px]"
          >
            Toast Title
          </div>
        </div>
        <div
          role="button"
        >
          <svg
            class="remixicon fill-slate-400 w-4 h-4"
            fill="default"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.9997 10.5865L16.9495 5.63672L18.3637 7.05093L13.4139 12.0007L18.3637 16.9504L16.9495 18.3646L11.9997 13.4149L7.04996 18.3646L5.63574 16.9504L10.5855 12.0007L5.63574 7.05093L7.04996 5.63672L11.9997 10.5865Z"
            />
          </svg>
        </div>
      </div>
      <div
        class="pl-4 flex items-start justify-center"
      >
        <div
          class="text-slate-500 text-sm whitespace-normal overflow-auto w-[300px]"
        >
          Text
        </div>
      </div>
    </div>
  </div>
</div>
`;
